queue_services/entity-filer/tests/unit/filing_processors/test_annual_report.py

```diff
@@ -14,36 +14,57 @@
 """The Test Suites to ensure that the worker is operating correctly."""
 import copy
 import datetime
+import pytest
 import random
 from unittest.mock import patch
 
 from freezegun import freeze_time
-from legal_api.models import Business, Filing
+from legal_api.models import BatchProcessing, Business, Filing
 from registry_schemas.example_data import ANNUAL_REPORT
 
 # from entity_filer.filing_processors.filing_components import create_party, create_role
 from entity_filer.filing_meta import FilingMeta
 from entity_filer.worker import process_filing
 from tests.unit import (
+    create_batch,
+    create_batch_processing,
     create_business,
     create_filing,
 )
+from tests import EPOCH_DATETIME
 
 
-def test_process_ar_filing(app, session):
+@pytest.mark.parametrize('test_name,flag_on,in_dissolution,eligibility,legal_type,', [
+    ('AR successfully', True, False, False, 'CP'),
+    ('Not withdrawn from the dissolution process', True, True, True, 'BC'),
+    ('Withdrawn from the dissolution process', True, True, False, 'CP'),
+    ('AR successfully when flag is off', False, True, False, 'CP')
+])
+def test_process_ar_filing_involuntary_dissolution(app, session, test_name, flag_on, in_dissolution, eligibility, legal_type):
     """Assert that an AR filling can be applied to the model correctly."""
-    from entity_filer.worker import APP_CONFIG
     from entity_filer.filing_processors import annual_report
     # vars
-    payment_id = str(random.SystemRandom().getrandbits(0x58))
     identifier = 'CP1234567'
 
-    # setup
-    business = create_business(identifier, 'CP')
-    business_id = business.id
-    now = datetime.date(2020, 9, 17)
-    ar_date = datetime.date(2020, 8, 5)
-    agm_date = datetime.date(2020, 7, 1)
+    business = create_business(identifier, legal_type)
+    business.founding_date = EPOCH_DATETIME
+    business.save()
+    # create the batch and batch_processing.
+    batch_status = 'PROCESSING'
+    if not in_dissolution:
+        batch_status = 'COMPLETED'
+    batch = create_batch(status=batch_status)
+    batch_processing = create_batch_processing(business=business, batch_id=batch.id, status=batch_status)
```

Comment:
When a business `in_dissolution` = False, I don't expect batch entries to exist for these tests.

Reply:
Updated